src/arith/int_set.cc

```diff
@@ -518,7 +533,14 @@ class IntervalSetEvaluator :
   // whether set is exactly single point that equals value.
   bool MatchPoint(const IntervalSet& set,
                   const PrimExpr& value) const {
-    return set->min_value.same_as(value) && set->max_value.same_as(value);
+    if (set->min_value.same_as(value) && set->max_value.same_as(value)) {
```

Comment:
Can we reuse `CanProve` here to check if `set->min_value == value && set->max_value == value`?

Reply:
Make sense.  I will keep the .same_as(value) calls for performance and use CanProve for the additional comparison.

Reply:
I don't need to change MatchPoint and IsSinglePoint after binding all ranges to one analyzer.  I have reverted the changes to these two functions.